sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs

```diff
@@ -558,18 +558,25 @@ private static SocketsHttpHandler ApplyOptionsToHandler(SocketsHttpHandler httpH
                 return httpHandler;
             }
 
+#pragma warning disable CA1416 // 'X509Certificate2' is unsupported on 'browser'
             // ServerCertificateCustomValidationCallback
             if (options.ServerCertificateCustomValidationCallback != null)
             {
                 httpHandler.SslOptions.RemoteCertificateValidationCallback = (_, certificate, x509Chain, sslPolicyErrors) =>
-#pragma warning disable CA1416 // 'X509Certificate2' is unsupported on 'browser'
                     options.ServerCertificateCustomValidationCallback(
                         new ServerCertificateCustomValidationArgs(
                             certificate is { } ? new X509Certificate2(certificate) : null,
                             x509Chain,
                             sslPolicyErrors));
-#pragma warning restore CA1416 // 'X509Certificate2' is unsupported on 'browser'
             }
+            // Set ClientCertificates
+             foreach (var cred in options.ClientCertificates)
+            {
+               httpHandler.SslOptions ??= new System.Net.Security.SslClientAuthenticationOptions();
```

Comment:
nit: add a using

Reply:
I think I did that on purpose here to avoid some type name collisions.